### PR TITLE
Triconfig - configure ui options overhaul

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: https://github.com/storybookjs/storybook/issues/8860

## What I did
- ADD handling for extra options set in `setConfig`
- CHANGE readme of addon-options, so we can remove it in 6.0.0
- ADD a deprecation warning when user import the options addon, so we can remove it in 6.0.0

### CODE REVIEW

File: addons/options/README.md [hunk 1]
```diff
@@ -1,136 +1,28 @@
...
-### Set options globally
-
-Import and use the `addParameters` + `options`-key in your `config.js` file.
+Global options can be specified with:
 
 ```js
-import { addParameters, configure } from '@storybook/react';
-
-// Option defaults:
-addParameters({
```
Comment (tmeasday) [inline addons/options/README.md]: We don't want to keep these JSdocs in this file (for the types of options?)

File: addons/options/README.md [hunk 2]
```diff
@@ -1,136 +1,28 @@
...
+import { themes } from '@storybook/theming';
+
+addons.setConfig({
+  theme: themes.dark,
+  showNav: true,
+  showPanel: true,
+  panelPosition: 'bottom',
+  enableShortcuts: true,
+  isToolshown: true,
+  selectedPanel: 'storybook/roundtrip',
```
Comment (tmeasday) [inline addons/options/README.md]: What happened to the other options (e.g. `url`, `showAddonPanel` etc)?

**Review by tmeasday — APPROVED** [review]
LGTM, some documentation qns

File: addons/options/README.md (hunk 2 above)
Comment (ndelangen) [reply addons/options/README.md]: They are to be documented on the docs site instead.

I want to completely remove this addon.

Comment (tmeasday) [reply addons/options/README.md]: Ok, but are these options documented somewhere else?

Comment (ndelangen) [reply addons/options/README.md]: https://github.com/storybookjs/storybook/blob/651e058cee6cecbb8280da4025c8e56345c883a3/docs/src/pages/configurations/options-parameter/index.md

Comment (ndelangen) [reply addons/options/README.md]: I updated to complete remove this content and just refer to the docs.

This prevents these docs from going out of sync.

File: addons/options/README.md (hunk 1 above)
Comment (ndelangen) [reply addons/options/README.md]: No, this was duplicated, I just refer to the docs site now.

**Pull request merged**